apps/andi/lib/andi_web/live/edit_live_view/data_dictionary_form.ex

```diff
@@ -80,12 +80,15 @@ defmodule AndiWeb.EditLiveView.DataDictionaryForm do
 
               <%= if @sourceFormat in ["text/csv", "application/json"] do %>
                 <div class="data-dictionary-form__file-upload">
-                  <div class="file-input-button">
-                    <%= label(f, :schema_sample, "Upload data sample", class: "label") %>
-                    <%= file_input(f, :schema_sample, phx_hook: "readFile", accept: "text/csv, application/json") %>
-                    <%= ErrorHelpers.error_tag(f, :schema_sample, bind_to_input: false) %>
+                  <div class="file-input-button--<%= loader_visibility %>">
+                    <div class="file-input-button">
+                      <%= label(f, :schema_sample, "Upload data sample", class: "label") %>
+                      <%= file_input(f, :schema_sample, phx_hook: "readFile", accept: "text/csv, application/json") %>
+                      <%= ErrorHelpers.error_tag(f, :schema_sample, bind_to_input: false) %>
+                    </div>
                   </div>
 
+                  <button type="button" id="reader-cancel" class="file-upload-cancel-button file-upload-cancel-button--<%= loader_visibility %> btn">Cancel</button>
```

Comment:
How does this flow down to the "change" event? Is it part of the `file_input` internals?

Reply:
There is an event listener in the `app.js` file, where the FileReader is being defined